bot/tests/test_reporter_phabricator.py

```diff
@@ -997,3 +998,36 @@ def test_phabricator_external_tidy(mock_phabricator, phab, mock_try_task, mock_t
 
     # Check the comment has been posted
     assert phab.comments[51] == [VALID_EXTERNAL_TIDY_MESSAGE]
+
+
+def test_phabricator_newer_diff(mock_phabricator, phab, mock_try_task):
+    """
+    Test Phabricator reporter publication won't be called when a newer diff exists for the patch
+    """
+
+    with mock_phabricator as api:
+        revision = Revision.from_try(mock_try_task, api)
+        revision.mercurial_revision = "deadbeef1234"
+        revision.repository = "https://hg.mozilla.org/try"
+        revision.repository_try_name = "try"
+        revision.lines = {
+            # Add dummy lines diff
+            "test.rst": [41, 42, 43],
+        }
+        reporter = PhabricatorReporter({"analyzers": ["doc-upload"]}, api=api)
+
+    os.environ["SPECIAL_NAME"] = "PHID-DREV-zzzzz-updated"
+    doc_url = "http://gecko-docs.mozilla.org-l1.s3-website.us-west-2.amazonaws.com/59dc75b0-e207-11ea-8fa5-0242ac110004/index.html"
+    doc_notice = COMMENT_LINK_TO_DOC.format(diff_id=42, doc_url=doc_url)
+    reporter.publish(
+        [],
+        revision,
+        [],
+        [doc_notice],
+    )
+
+    # Check the comment hasn't been posted
+    assert phab.comments[51] == []
+
```

Comment:
Can you make sure the issues are still published as Harbormaster results?

Reply:
(I mean as part of the test)

Reply:
done!